app/src/main/cpp/skyline/common.h

```diff
@@ -194,34 +180,127 @@ namespace skyline {
             if (hexString.size() != Size * 2)
                 throw exception("Invalid size");
             std::array<u8, Size> result;
-            for (size_t i{}; i < Size; ++i) {
+            for (size_t i{}; i < Size; i++) {
                 size_t hexStrIndex{i * 2};
                 result[i] = (HexDigitToByte(hexString[hexStrIndex]) << 4) | HexDigitToByte(hexString[hexStrIndex + 1]);
             }
             return result;
         }
 
+        /**
+         * @brief A compile-time hash function as std::hash isn't constexpr
+         */
         constexpr std::size_t Hash(std::string_view view) {
             return frz::elsa<frz::string>{}(frz::string(view.data(), view.size()), 0);
         }
+    }
+
+    /**
+     * @brief A custom wrapper over span that adds several useful methods to it
+     * @note This class is completely transparent, it implicitly converts from and to span
+     */
+    template<typename T, size_t Extent = std::dynamic_extent>
+    class span : public std::span<T, Extent> {
+      public:
+        using std::span<T, Extent>::span;
+        using std::span<T, Extent>::operator=;
+
+        typedef typename std::span<T, Extent>::element_type elementType;
+        typedef typename std::span<T, Extent>::index_type indexType;
+
+        constexpr span(const std::span<T, Extent> &spn) : std::span<T, Extent>(spn) {}
+
+        /**
+         * @brief We want to support implicitly casting from std::string_view -> span as it is just a specialization of a data view which span is a generic form of, the opposite doesn't hold true as not all data held by a span is string data therefore the conversion isn't implicit there
+         */
+        template<class Traits>
+        constexpr span(const std::basic_string_view<T, Traits> &string) : std::span<T, Extent>(const_cast<T *>(string.data()), string.size()) {}
+
+        template<typename Out>
+        constexpr Out &as() {
+            if (span::size_bytes() >= sizeof(Out))
+                return *reinterpret_cast<Out *>(span::data());
+            throw exception("Span size is less than Out type size (0x{:X}/0x{:X})", span::size_bytes(), sizeof(Out));
+        }
 
-        template<typename Out, typename In>
-        constexpr Out &As(std::span<In> span) {
-            if (IsAligned(span.size_bytes(), sizeof(Out)))
-                return *reinterpret_cast<Out *>(span.data());
-            throw exception("Span size not aligned with Out type size (0x{:X}/0x{:X})", span.size_bytes(), sizeof(Out));
+        constexpr std::string_view as_string(indexType length = 0) {
+            return std::string_view(reinterpret_cast<char *>(span::data()), length ? length : span::size_bytes());
```

Comment:
Can we make this get a string up to the `max(find(first null terminator), length)`, otherwise it gets kinda annoying with comparison as the null bytes at the end count as part of the string.

Reply:
Just know that I'll have to wait for the entire project to recompile now that I have to make a single modification to `common.h`

Reply:
I was correcting your capitalization